tests/e2e/vm_service_utils.go

```diff
@@ -0,0 +1,586 @@
+	vols := []vmopv1.VirtualMachineVolume{}
+	vmName := fmt.Sprintf("csi-test-vm-%d", r.Intn(10000))
+	for _, pvc := range pvcs {
+		vols = append(vols, vmopv1.VirtualMachineVolume{
+			Name: pvc.Name,
+			PersistentVolumeClaim: &vmopv1.PersistentVolumeClaimVolumeSource{
+				PersistentVolumeClaimVolumeSource: v1.PersistentVolumeClaimVolumeSource{ClaimName: pvc.Name},
+			},
+		})
+	}
+	vm := vmopv1.VirtualMachine{
+		ObjectMeta: metav1.ObjectMeta{Name: vmName, Namespace: namespace},
+		Spec: vmopv1.VirtualMachineSpec{
+			PowerState:   "poweredOn",
+			ImageName:    vmi,
+			ClassName:    vmClassBestEffortSmall,
+			StorageClass: storageClassName,
+			Volumes:      vols,
+			VmMetadata:   &vmopv1.VirtualMachineMetadata{Transport: "CloudInit", SecretName: secretName},
+		},
+	}
+	err := c.Create(ctx, &vm)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	return waitNgetVmsvcVM(ctx, c, namespace, vmName)
+}
+
+// getVmsvcVM fetches the vm from the specified ns
+func getVmsvcVM(
+	ctx context.Context, c ctlrclient.Client, namespace string, vmName string) (*vmopv1.VirtualMachine, error) {
+	instanceKey := ctlrclient.ObjectKey{Name: vmName, Namespace: namespace}
+	vm := &vmopv1.VirtualMachine{}
+	err := c.Get(ctx, instanceKey, vm)
+	return vm, err
+}
+
+// waitNgetVmsvcVM wait and fetch the vm CR from the specified ns
+func waitNgetVmsvcVM(ctx context.Context, c ctlrclient.Client, namespace string, vmName string) *vmopv1.VirtualMachine {
+	vm := &vmopv1.VirtualMachine{}
+	var err error
+	err = wait.PollImmediate(poll*5, pollTimeout, func() (bool, error) {
+		vm, err = getVmsvcVM(ctx, c, namespace, vmName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return false, err
+			}
+			return false, nil
+		}
+		return true, nil
+	})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	return vm
+}
+
+// waitNgetVmsvcVmIp wait and fetch the primary IP of the vm in give ns
+func waitNgetVmsvcVmIp(ctx context.Context, c ctlrclient.Client, namespace string, name string) string {
+	ip := ""
+	err := wait.PollImmediate(poll*10, pollTimeout*2, func() (bool, error) {
+		vm, err := getVmsvcVM(ctx, c, namespace, name)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return false, err
+			}
+			return false, nil
+		}
+		if vm.Status.VmIp == "" {
+			return false, nil
+		}
+		ip = vm.Status.VmIp
+		return true, nil
+	})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	framework.Logf("Found IP '%s' for VM '%s'", ip, name)
+	return ip
+}
+
+// createBootstrapSecretForVmsvcVms create bootstrap data for cloud init in the ns
+func createBootstrapSecretForVmsvcVms(ctx context.Context, client clientset.Interface, namespace string) string {
+	secretSpec := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: "vm-bootstrap-data", Namespace: namespace},
+		StringData: map[string]string{"user-data": `#cloud-config
+ssh_pwauth: true
+users:
+- default
+- name: worker
+  lock_passwd: false
+  plain_text_passwd: 'ca$hc0w'
```

Comment:
should we parameterize passwd and not hardcode ?

Reply:
since this is an internal network vm, it is not accessible to outside n/w
have tried to keep things simple